web/src/hashchange.js

```diff
@@ -55,7 +55,10 @@ function show_all_message_view() {
     const coming_from_recent_view = maybe_hide_recent_view();
     const coming_from_inbox = maybe_hide_inbox();
     const is_actively_scrolling = message_scroll.is_actively_scrolling();
-    narrow.deactivate(!(coming_from_recent_view || coming_from_inbox), is_actively_scrolling);
+    narrow.deactivate(
+        !(coming_from_recent_view || coming_from_inbox || narrow.has_visited_all_messages),
```

Comment:
Hmm, the `narrow.has_visited_all_messages` check here is I think inverted? It's hard to think about, but the point is that you can't be coming from all messages if you haven't been there before.

But the logic here is also wrong for the name `coming_from_all_messages`. I think the right fix is to rename `coming_from_all_messages` to an opposite-meaning `message_feed_previously_hidden = coming_from_recent_view || coming_from_inbox)`

Reply:
Got it. I think it actually is correct, because I'm initializing the `has_visited_all_messages` value to `false`, which means it'll be false when `show_all_message_view()` is called on a reload, which is negated to `true` when that runs...and will therefore also be false inside of `narrow.deactivate()` when it runs the first time.

But agreed--there's room to make this way less twisty-turny. I'll pick this up with fresher eyes and brain on Monday.